### PR TITLE
bug_fix: when the number of ep.Subsets changes from non-zero to zero, the sync logic will not run

### PR DESCRIPTION
**Pull request by flying1pig**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix

- Related issues
https://github.com/apache/apisix-ingress-controller/issues/464
___
### Bugfix
- Description
when the number of ep.Subsets changes from non-zero to zero, the endpoint controller will not run into sync logic which sync changes from apisix-ingress-controller to apisix.

- How to fix?
if the length of ep.Subsets equals zero, sync empty apisixv1.UpstreamNode to apisix.

___
### New feature or improvement
- Describe the details and related test reports.
step1. make sure that the apisix-ingress-controller is working correctly with k8s and apisix cluster;
step2. chose a endpoint object which is listening by apisix-ingress-controller, change the related replicas from non-zero to zero;
step3. the apisix dashboard will not update the changes.  

___
### Backport patches
- Why need to backport?
it's a bug in all of the history branch.

- Source branch
master

- Related commits and pull requests
no

- Target branch
all

### CODE REVIEW

File: pkg/ingress/endpoint.go [hunk 1]
```diff
@@ -114,6 +114,21 @@ func (c *endpointsController) sync(ctx context.Context, ev *types.Event) error {
 		portMap[port.Name] = port.Port
 	}
 	clusters := c.controller.apisix.ListClusters()
+
+	//endpoints turn to zero
+	if len(ep.Subsets) == 0 {
+		nodes := []apisixv1.UpstreamNode{}
```
Comment (tokers) [inline pkg/ingress/endpoint.go:120]: We may still need to respect the ApisixUpstream subset settings, which means we need to traverse the ApisixUpstream subsets and reset the nodes to empty for all of them.

**Comment by gxthrj** [comment]
@flying1pig any update?

**Comment by tao12345666333** [comment]
@flying1pig friendly reminder

**Comment by tao12345666333** [comment]
@flying1pig Can you resolve the conflict?

**Comment by tao12345666333** [comment]
#1140 has been merged. It can cover this case. Let's close this one. Thanks

**Pull request closed without merging**